pkg/volume/git_repo/git_repo.go

```diff
@@ -276,3 +294,10 @@ func getVolumeSource(spec *volume.Spec) (*v1.GitRepoVolumeSource, bool) {
 
 	return volumeSource, readOnly
 }
+
+func validateNonFlagArgument(arg, argName string) error {
+	if len(arg) > 0 && arg[0] == '-' {
```

Comment:
do we need to trim first? would git be fooled by " --recursive"?

Reply:
Tested this manually, and it just treats it as a directory named " --recursive".